src/shapes/path.cpp

```diff
@@ -1,15 +1,26 @@
 #include "rive/shapes/path.hpp"
-#include "rive/math/circle_constant.hpp"
 #include "rive/renderer.hpp"
 #include "rive/shapes/cubic_vertex.hpp"
 #include "rive/shapes/cubic_detached_vertex.hpp"
 #include "rive/shapes/path_vertex.hpp"
 #include "rive/shapes/shape.hpp"
 #include "rive/shapes/straight_vertex.hpp"
+#include "rive/math/math_types.hpp"
 #include <cassert>
 
 using namespace rive;
 
+/// Compute an ideal control point distance to create a curve of the given
+/// radius.
+static float
+computeIdealControlPointDistance(const Vec2D& toPrev, const Vec2D& toNext, float radius) {
+    // Get the angle between next and prev
+    float angle = fabs(atan2(Vec2D::cross(toPrev, toNext), Vec2D::dot(toPrev, toNext)));
+
+    return (4.0f / 3.0f) * tan(math::PI / (2.0f * ((2.0f * math::PI) / angle))) * radius *
```

Comment:
How did you arrive at this equation? (The commit message says it's a guess/heuristic?) If you're trying to approximate a circular arc using a cubic, there is an exact optimal solution here:

https://github.com/google/skia/blob/main/samplecode/SampleFitCubicToCircle.cpp#L68

Reply:
As discussed offline:

https://observablehq.com/@daformat/rounding-polygon-corners

This algorithm look much better than circular!

Reply:
Actually, can you add that link to the comments?

https://github.com/google/skia/blob/main/samplecode/SampleFitCubicToCircle.cpp#L68

Reply:
Actually, can you mention "natural rounding" and add that link to the comments?

https://observablehq.com/@daformat/rounding-polygon-corners